### PR TITLE
juju 3.0 renames the runner binary

### PR DESCRIPTION
**Pull request by addyess**
This service responds to a systemctl daemon which watches a set of files on the leader unit, and syncs them to other units in the cluster.

These changes ensure one of the binaries is appropriately called

Fixes: [LP:2006977](https://bugs.launchpad.net/charm-kubernetes-master/+bug/2006977)

### CODE REVIEW

File: .wokeignore
Comment (kwmonroe) [inline .wokeignore:4]: TIL of this `.wokeignore` syntax.  neat!

File: templates/cdk.master.leader.file-watcher.service
Comment (kwmonroe) [inline templates/cdk.master.leader.file-watcher.service:8]: TIL of this `=-` syntax. neat!

**Review by kwmonroe — APPROVED** [review]

File: templates/cdk.master.leader.file-watcher.service
Comment (addyess) [reply templates/cdk.master.leader.file-watcher.service:8]: my only concern here is that `ExecStart=-/usr/bin/bad_command` will just log any errors but the service won't fail.  So, there will be no reporting of this service ACTUALLY failing if juju decides to rename a command again or alias the command.  

if they alias it (ie -- both binaries are present on the system) -- well then, BOTH commands would be run -- effectively calling this process twice. 

i also considered

```suggestion
ExecStart= \
  /usr/bin/juju-exec {{ unit }} /usr/local/sbin/cdk.master.leader.file-watcher.sh || 
  /usr/bin/juju-run {{ unit }} /usr/local/sbin/cdk.master.leader.file-watcher.sh
```

File: .wokeignore
Comment (addyess) [reply .wokeignore:4]: this is actually the syntax to ignore a single baddy name... i'm actually woke-ignoring the `.wokeignore` file here.

File: templates/cdk.master.leader.file-watcher.service
Comment (addyess) [reply templates/cdk.master.leader.file-watcher.service:8]: hmm... something about the failed integration tests and these changes troubles me..

File: templates/cdk.master.leader.file-watcher.service
Comment (kwmonroe) [inline templates/cdk.master.leader.file-watcher.service:9]: @addyess how about this to address your concern with both `juju-[exec|run]` failing?  kinda ugly, but `bash -c` was the simplest way i found to chain ExecStarts.

Confirmed with edge charms + this PR:
```
ubuntu@juju-8ac7ee-6:~$ systemctl status cdk.master.leader.file-watcher.service 
○ cdk.master.leader.file-watcher.service - CDK control-plane leader file-watcher
     Loaded: loaded (/etc/systemd/system/cdk.master.leader.file-watcher.service; disabled; vendor preset: enabled)
     Active: inactive (dead) since Mon 2023-02-13 18:02:50 UTC; 17s ago
TriggeredBy: ● cdk.master.leader.file-watcher.path
    Process: 1482911 ExecStart=/bin/bash -c /usr/bin/juju-exec kubernetes-control-plane/1 /usr/local/sbin/cdk.master.leader.file-watcher.sh ||      /usr/bin/juju-run kubernetes-contr>
   Main PID: 1482911 (code=exited, status=0/SUCCESS)
        CPU: 78ms

Feb 13 18:02:50 juju-8ac7ee-6 systemd[1]: Starting CDK control-plane leader file-watcher...
Feb 13 18:02:50 juju-8ac7ee-6 bash[1482912]: /bin/bash: line 1: /usr/bin/juju-exec: No such file or directory
Feb 13 18:02:50 juju-8ac7ee-6 systemd[1]: cdk.master.leader.file-watcher.service: Deactivated successfully.
Feb 13 18:02:50 juju-8ac7ee-6 systemd[1]: Finished CDK control-plane leader file-watcher.
```

Comment (kwmonroe) [reply templates/cdk.master.leader.file-watcher.service:9]: Welp, this isn't holding up much better; looks like we're going for the 6hr timeout on tests.

Comment (addyess) [reply templates/cdk.master.leader.file-watcher.service:9]: this wasn't the reason behind the test-timeouts.  Turns out i had the wrong version of python libjuju selected in tox.ini.  I should have constrained it to `juju < 3.1`.  The 3.0 version of `juju` library has improved async handling on run events -- the very first `juju_run` call in the test was the point of the failure

File: templates/cdk.master.leader.file-watcher.sh
Comment (addyess) [reply templates/cdk.master.leader.file-watcher.sh]: sure enough
```bash
❯ juju run -u kubernetes-control-plane/1 -- 'is-leader; echo $?'
False
0

```

**Review by kwmonroe — APPROVED** [review]
re-approved again once more ftw

**Pull request merged**
